tests/unit_tests/data_steward/retraction/create_deid_map_test.py

```diff
@@ -0,0 +1,125 @@
+            expected_all_datasets, expected_deid_datasets)
+
+        self.assertListEqual(expected_combined_datasets, result)
+
+    def test_get_corresponding_combined_dataset(self):
+        result = create_deid_map.get_corresponding_combined_dataset(
+            self.all_datasets, self.deid_datasets)
+        expected = list()
+
+        for d in self.deid_datasets:
+            release = d.split('_')[0]
+            older_release = release[1:]
+            combined, older_combined = release + '_combined', older_release + '_combined'
+
+            if combined in self.all_datasets:
+                expected.append(combined)
+            if older_combined in self.all_datasets:
+                expected.append(older_combined)
+            # Remove duplicates
+        expected = list(set(expected))
+
+        self.assertEquals(expected, result)
+
+    @mock.patch('utils.bq.get_table_info_for_dataset')
+    def test_check_if_deid_map_exists(self, mock_get_table_info):
+        result = create_deid_map.check_if_deid_map_exists(
+            self.project_id, self.dataset_id)
+
+        expected_table_info_df = mock_get_table_info.return_value = pd.DataFrame(
+            columns=[
+                'table_catalog', 'table_schema', 'table_name', 'column_name',
+                'ordinal_position', 'is_nullable', 'data_type', 'is_generated',
+                'generation_expression', 'is_stored', 'is_hidden',
+                'is_updatable', 'is_system_defined', 'is_partitioning_column',
+                'clustering_ordinal_position'
+            ])
+        expected_column_list = expected_table_info_df['table_name'].tolist()
+
+        if 'deid_map' in expected_column_list:
+            self.assertEquals('rename required', result)
+        if '_deid_map' in expected_column_list:
+            self.assertEquals(True, result)
+        if ['deid_map', '_deid_map'] not in expected_column_list:
+            self.assertEquals(False, result)
+
+    def test_rename_deid_map_table_query(self):
+        result = create_deid_map.rename_deid_map_table_query(
+            self.project_id, self.dataset_id)
+        expected = create_deid_map.RENAME_DEID_MAP_TABLE_QUERY.format(
+            project=self.project_id, dataset=self.dataset_id)
+        self.assertEquals(result, expected)
+
+    def test_create_deid_map_table_query(self):
+        result = create_deid_map.create_deid_map_table_query(
+            self.project_id, self.dataset_id)
+        expected = create_deid_map.CREATE_DEID_MAP_TABLE_QUERY.format(
+            project=self.project_id, dataset=self.dataset_id)
+        self.assertEquals(result, expected)
+
+    @mock.patch('retraction.create_deid_map.utils.bq.list_datasets')
+    def test_create_deid_map_table_queries(self, mock_list_datasets):
+        result = create_deid_map.create_deid_map_table_queries(self.project_id)
+        mock_list_datasets.return_value = utils.bq.list_datasets(
+            self.project_id)
+
```

Comment:
You could supply this with a list like 
```suggestion
        mock_list_datasets.return_value = self.all_datasets
```
and see what queries it generates. If it is as expected depending on the existence of combined datasets for each deid datasets, it is successful. Also, you might want to mock `get_table_info_for_dataset` such that it returns a data frame with tables as in the test above. Since it will be called as many times as the number of combined datasets with corresponding deid datasets, you could return it different dataframes based on the call count. This can be achieved by something like
```suggestion
        mock_list_datasets.return_value = self.all_datasets
        mock_table_info.side_effect = [dataframe_1, dataframe_2, dataframe_3]
```
where one could have no deid map, another could have `deid_map` and another `_deid_map`.

Reply:
thank you, I think I've updated this correctly!